heat/core/linalg.py

```diff
@@ -877,3 +877,87 @@ def triu(m, k=0):
         Upper triangle of the input tensor.
     """
     return __tri_op(m, k, torch.triu)
+
+def norm(w,norm_order=2):
+    """
+        Calculates the norm of a given vector
+    """
+    beta = abs(max(w[0], key=abs))
+    if beta == 0:
+        beta = 1
+    v = [element/beta for element in w[0]]
+
+    v_norm = (sum(element**norm_order for element in v))**(1/norm_order)
+    x_norm = v_norm * beta
+
+    return x_norm
+
+def householder_reflect(w):
+    """
+        Calculates the reflection of a vector w using the householder reflector
+    """
+    tau = norm(w.T)
+
+    if tau == 0 or w.shape[0] == 1:
+        return 0, np.zeros((w.shape)), 0, np.identity((w.shape[0]))
```

Comment:
There should not be any numpy code here. Instead it should (preferably) be ht.zeros and ht.eye calls or at least torch.zeros and torch.eye to allow GPU computations